console/services/market_app_service.py

```diff
@@ -1341,33 +1356,41 @@ def delete_rainbond_app_version(self, enterprise_id, app_id, version):
             raise e
 
     def get_rainbond_app_and_versions(self, enterprise_id, app_id, page, page_size):
-        have_version = False
         app = rainbond_app_repo.get_rainbond_app_by_app_id(enterprise_id, app_id)
-        app_versions = rainbond_app_repo.get_rainbond_app_versions_by_id(enterprise_id, app_id)
         if not app:
             raise RbdAppNotFound("未找到该应用")
+        app_versions = rainbond_app_repo.get_rainbond_app_version_by_app_ids(enterprise_id, [app_id]).values()
+
+        apv_ver_nums = []
         app_release = False
-        if app_versions is not None:
-            for version in app_versions:
-                if version["version"]:
-                    have_version = True
-                version["release_user"] = ""
-                version["share_user_id"] = version["share_user"]
-                version["share_user"] = ""
-                user = Users.objects.filter(user_id=version["release_user_id"]).first()
-                share_user = Users.objects.filter(user_id=version["share_user_id"]).first()
-
-                if user:
-                    version["release_user"] = user.nick_name
-                if share_user:
-                    version["share_user"] = share_user.nick_name
-                else:
-                    record = app_import_record_repo.get_import_record(version["record_id"])
-                    if record:
-                        version["share_user"] = record.user_name
-                version["dev_status"] = version.dev_status
-                if version["dev_status"] == "release":
-                    app_release = True
+        app_with_versions = dict()
+        for version in app_versions:
+            if version["dev_status"] == "release":
+                app_release = True
+
+            version["release_user"] = ""
+            version["share_user_id"] = version["share_user"]
+            version["share_user"] = ""
+            user = Users.objects.filter(user_id=version["release_user_id"]).first()
```

Comment:
if the user not exists, will raise an exception?

Reply:
No,refer to:
> first()¶
Returns the first object matched by the queryset, or None if there is no matching object. If the QuerySet has no ordering defined, then the queryset is automatically ordered by the primary key. This can affect aggregation results as described in Interaction with default ordering or order_by().